client/state/data-layer/wpcom/concierge/schedules/shifts/index.js

```diff
@@ -0,0 +1,52 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { translate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { updateConciergeShifts } from 'state/concierge/actions';
+import { errorNotice } from 'state/notices/actions';
+import { CONCIERGE_SHIFTS_REQUEST } from 'state/action-types';
+
+export const fetchConciergeShifts = ( { dispatch }, action ) => {
+	const { scheduleId } = action;
+
+	dispatch(
+		http(
+			{
+				method: 'GET',
+				path: `/concierge/schedules/${ scheduleId }/shifts`,
+				apiNamespace: 'wpcom/v2',
+			},
+			action
+		)
+	);
+};
+
+export const storeFetchedConciergeShifts = ( { dispatch }, action, shifts ) =>
+	dispatch( updateConciergeShifts( shifts ) );
+
+export const showConciergeShiftsFetchError = ( { dispatch } ) =>
+	dispatch(
+		errorNotice(
+			translate(
+				"We've encountered problems trying to load available shifts. Please try again later."
```

Comment:
I don't think this is the best error message to show customers, who are landing into the Concierge scheduler with no context about what a "shift" is. Maybe instead:

> We couldn't load our Concierge schedule. Please try again later.

Reply:
Ah, that's better :)
Updated in 4e9f5d1.